pwncat/manager.py

```diff
@@ -623,6 +629,16 @@ def create_session(self, platform: str, channel: Channel = None, **kwargs):
         """
 
         session = Session(self, platform, channel, **kwargs)
+
+        # increment the ``session_id`` variable upon adding a new session
+        # this may cause issues when a session is not properly configured
+        # for example when the victim machine disconnects or terminates the parent process
+        # in that case, we have already incremented the ``session_id``
+        # thus the ``session_id`` for the bad session is no longer in use
+        # check other modules to prevent this from happening
+
+        self.session_id += 1
```

Comment:
Apologies for above comments, I added them for my own work

Currently, interacting with an invalid session (_disconnected_) throws an _error_ and shows the _traceback_
I will try to add a `try-except-finally` block to capture this and show a simple error message instead of showing the whole _traceback_ to the user

Reply:
In what case do exceptions occur? I didn't see any exceptions while testing the `sessions` command on your fork.

Reply:
There is not much effect on a _single session_ use case other than the display of _traceback_.
But when there are _multiple sessions_, interacting with an _invalid/disconnected_ session causes the `pwncat` instance to exit out

Reply:
In what specific cases are you getting a traceback? The check at line 75 of `pwncat/commands/sessions.py` *should* stop that, should it not? I tested with your fork, and wasn't able to trigger a traceback:

![image](https://user-images.githubusercontent.com/7529189/122269079-cd5b3180-ceaa-11eb-9bd2-7a525817c982.png)

Reply:
Allow me some time to reproduce the issue, I do have a _screenshot_ from my earlier tests when I _encountered_ this issue.

Reply:
Here I was fuzzing around with `Ctrl-D` and the response times for different shells.
I killed `socat` in order to _invalidate_ two of the _existing sessions_

![Screenshot from 2021-06-16 21-08-23](https://user-images.githubusercontent.com/44923300/122269709-22269880-cefb-11eb-98e4-3a61e2e551ce.png)

Just after the above actions, I tried to interact with the _invalidated session_ - `sessions 0` and waited until I got kicked out :sweat_smile: 

![Screenshot from 2021-06-16 21-09-02](https://user-images.githubusercontent.com/44923300/122269716-23f05c00-cefb-11eb-8faa-18a79920655f.png)

Reply:
When we consider `KoTH` from [_TryHackMe_](https://tryhackme.com/games/koth), our opponents sometimes kill our shell.
In order to introduce a work, I used `sleep 10` in the first image.

Reply:
Ah, okay. That's an issue with the `Session.close`. It should catch `ChannelError` and `PlatformError` when calling `self.platform.exit()` and `self.platform.channel.close()`. Probably something like this would be appropriate in `Session.close`:

```py
try:
  self.platform.exit()
  self.platform.channel.close()
except (ChannelError, PlatformError) as exc:
  self.log(f"[yellow]warning[/yellow]: exception caught while closing channel: {exc}")
```

Reply:
I haven't tested your suggestion yet, but it looks promising.
I will try `try-except`, moreover there were some more _exceptions_ being __raised__ regarding some _connection errors_ which can be easily suppressed from the end-user by the use of `exception-handling` and showing appropriate error message

I have some suggestions as well regarding some features, I don't personally think it necessary to create issues (literally _Issues_) for them as of now. How may I share them with you? Or should I create __the Issue__?

Reply:
Yeah, the fix for the exceptions may be a larger thing than what this pull request originally was. The exceptions problem isn't really related to your changes to the session ID logic or small f-string fixes. As such, you can hold off on those, if you'd like to just get this merged in.

Regarding other feature ideas, if you'd like to open a new "Feature Request" issue, that's totally valid. If you're unsure about it, you can also use the "Discussions" tab at the top of the repo until you feel the idea is fleshed out.

Reply:
Yes, I did use __Currently, interacting...__ in my comment :sweat_smile: 
I should have been more _verbose_ there, that it is not about my _PR_